src/jmh/java/software/amazon/cloudwatchlogs/emf/MetricsLoggerBenchmark.java

```diff
@@ -0,0 +1,345 @@
+package software.amazon.cloudwatchlogs.emf;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+import org.openjdk.jmh.annotations.*;
+import software.amazon.cloudwatchlogs.emf.environment.Environment;
+import software.amazon.cloudwatchlogs.emf.environment.EnvironmentProvider;
+import software.amazon.cloudwatchlogs.emf.logger.MetricsLogger;
+import software.amazon.cloudwatchlogs.emf.model.DimensionSet;
+import software.amazon.cloudwatchlogs.emf.sinks.SinkShunt;
+
+@State(Scope.Benchmark)
+@BenchmarkMode(Mode.AverageTime)
+@Warmup(iterations = 3, time = 5)
+@Measurement(iterations = 3, time = 5)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@Fork(value = 1)
+public class MetricsLoggerBenchmark {
+    private MetricsLogger logger;
+    private EnvironmentProvider envProvider;
+    private SinkShunt sink;
+    private Environment environment;
+
+    @Setup
+    public void setUp() {
+        envProvider = mock(EnvironmentProvider.class);
+        environment = mock(Environment.class);
+        sink = new SinkShunt();
+
+        when(envProvider.resolveEnvironment())
+                .thenReturn(CompletableFuture.completedFuture(environment));
+        when(environment.getSink()).thenReturn(sink);
+        logger = new MetricsLogger(envProvider);
+    }
+
+    /**
+     * Publishing 10000 metrics with single thread. no lock: 0.844 ms/op; RW lock: 0.896 ms/op; S
```

Comment:
let's put this result as a table and put it as comment in this PR with background on what you are testing. otherwise when people read this in the future they won't understand what does S lock mean

Reply:
Sure! So I can delete the comments on benchmarking results once I have the table, am I understanding it right?